pipenv/core.py

```diff
@@ -1889,12 +1889,16 @@ def do_install(
     # Capture -e argument and assign it to following package_name.
     more_packages = list(more_packages)
     if package_name == '-e':
+        if not more_packages:
+            raise click.BadArgumentUsage('Please provide path to setup.py')
         package_name = ' '.join([package_name, more_packages.pop(0)])
     # capture indexes and extra indexes
     line = [package_name] + more_packages
     index_indicators = ['-i', '--index', '--extra-index-url']
     index, extra_indexes = None, None
     if more_packages and any(more_packages[0].startswith(s) for s in index_indicators):
+        if len(more_packages) < 2:
+            raise click.BadArgumentUsage('Please provide index value')
```

Comment:
I feel the logic in this block can be rewritten a bit to make the checks clearer. Right now the various checks are a bit convoluted, and require a lot of context to be understood. Why does it use `more_packages` here, but `package_names` below? How do they relate to `line`?

Reply:
@uranusjr I agree with you about some extra context what you should take in mind. I'll try to rewrite this block.